components/mailchimp/sources/new-campaign/new-campaign.js

```diff
@@ -0,0 +1,101 @@
+const common = require("../common/timer-based");
+
+module.exports = {
+  ...common,
+  key: "new-campaign",
+  name: "New Campaign",
+  description: "Whether to emit events when campaigns are created or when they are sent.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    type: {
+      type: "string",
+      label: "type",
+      description:
+        "There are four types of campaigns you can create in Mailchimp. A/B Split campaigns have been deprecated and variate campaigns should be used instead. ",
+      options: [
+        "regular",
+        "plaintext",
+        "absplit",
+        "rss",
+        "variate",
+      ],
+      optional: false,
+    },
+    status: {
+      type: "string",
+      label: "status",
+      description: "Campaign status to watch for.",
+      options: [
+        "created",
+        "sent",
+      ],
+      default: "created",
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      const config = {
+        count: 10,
+        offset: 0,
+      };
+      let campaigns;
+      if(this.mailchimp.statusIsSent(this.status)) {
+        campaigns = await this.mailchimp.getCampaignsBySentDate(config);
+      } else {
+        campaigns = await this.mailchimp.getCampaignsByCreationDate(config);
+      }
```

Comment:
Could simplify it this way:
```suggestion
      const campaigns = this.mailchimp.statusIsSent(this.status) 
        ? await this.mailchimp.getCampaignsBySentDate(config) 
        : await this.mailchimp.getCampaignsByCreationDate(config);
```